src/main/java/io/confluent/idesidecar/restapi/models/graph/RealDirectFetcher.java

```diff
@@ -93,17 +93,21 @@ public Uni<DirectKafkaCluster> getKafkaCluster(String connectionId) {
     if (spec != null && DIRECT.equals(spec.type()) && spec.kafkaClusterConfig() != null) {
       var kafkaConfig = spec.kafkaClusterConfig();
       var ccloudEndpoint = kafkaConfig.asCCloudEndpoint();
+      var clusterId = orDefault(
+          kafkaConfig.id(), () -> connectionId + "-kafka-cluster"
+      );
       if (ccloudEndpoint.isPresent()) {
         var endpoint = ccloudEndpoint.get();
         cluster = new DirectKafkaCluster(
-            endpoint.clusterId().toString(),
-            spec.name() + " Kafka Cluster",
-            uriUtil.getHostAndPort(endpoint.getUri()),
+            // TODO: Use the cluster ID from connecting with admin client cluster-describe request
+            clusterId, //endpoint.clusterId().toString(),
```

Comment:
Do we need this code-in-comment?

```suggestion
            clusterId,
```

Reply:
Yes, I'd like to keep it, because this is something that has to be addressed with #123. See [my comment above](https://github.com/confluentinc/ide-sidecar/pull/152#issuecomment-2471808862):
> the `kafka_cluster.id` and `schema_registry.id` are currently optional in the OpenAPI spec but are required until we can obtain the cluster ID of the remote system and verify it matches. The `RealDirectFetcher` will need to perform a describe-cluster using the admin client, and set the cluster ID. (We might consider remove the `kafka_cluster.id` field and `schema_registry.id` field from the connection spec, if we always get a good ID from the describe-cluster.)